### PR TITLE
Add the possibility to redirect specify smartcard readers - 2nd try

### PR DESCRIPTION
**Pull request by HenryJacques**
Same PR as #2959 
But this time it should be clean ;)

### CODE REVIEW

**Comment by akallabeth** [comment]
@HenryJacques Sorry, still not up to date with current master...

File: client/common/cmdline.c [hunk 1]
```diff
@@ -830,15 +831,26 @@ static int freerdp_client_command_line_post_filter(void* context, COMMAND_LINE_A
...
-
-		free(p);
+		if (arg->Flags & COMMAND_LINE_VALUE_PRESENT)
+		{
+			char** p;
+			int count;
+			
+			p = freerdp_command_line_parse_comma_separated_values_offset(arg->Value, &count);
+			p[0] = "smartcard";
+			free(p);
```
Comment (akallabeth) [inline client/common/cmdline.c]: @HenryJacques This is definitely broken, you do not load the channel and free the arguments...

Comment (HenryJacques) [inline client/common/cmdline.c]: @akallabeth  What is broken? On my computer the smartcard channel gets loaded twice (as always) and /smartcard & /smartcard: have the same behaviour.

**Comment by akallabeth** [comment]
the pull is still based on an incompatible branch and when adding some arguments to /smartcard you will enter the branch not loading the channel ;)

**Comment by HenryJacques** [comment]
@akallabeth I'm really sorry to ask this, but how do you know it is still based on an incompatible branch... ?
The files diff (https://github.com/FreeRDP/FreeRDP/pull/3126/files) only show my changes.
And I can guarantee that the channel is loading.

**Comment by HenryJacques** [comment]
@akallabeth nevermind you're right :/
I'll change that.

**Comment by HenryJacques** [comment]
Cancelled

**Pull request closed without merging**